diff --git a/src/gallium/targets/wgl/gallium_wgl.mingw.def b/src/gallium/targets/wgl/gallium_wgl.mingw.def
index ee23392d95e..ef075870a5c 100644
--- a/src/gallium/targets/wgl/gallium_wgl.mingw.def
+++ b/src/gallium/targets/wgl/gallium_wgl.mingw.def
@@ -24,3 +24,12 @@ EXPORTS
 	stw_pixelformat_choose = stw_pixelformat_choose@8
 	stw_pixelformat_get = stw_pixelformat_get@4
 	stw_override_opengl32_entry_points = stw_override_opengl32_entry_points@8
+	stw_create_context_attribs = stw_create_context_attribs@32
+	stw_destroy_context = stw_destroy_context@4
+	stw_unbind_context = stw_unbind_context@4
+	stw_current_context = stw_current_context@0
+	stw_make_current = stw_make_current@12
+	stw_framebuffer_create = stw_framebuffer_create@12
+	stw_framebuffer_release_locked = stw_framebuffer_release_locked@4
+	stw_framebuffer_unlock = stw_framebuffer_unlock@4
+	stw_framebuffer_swap_locked = stw_framebuffer_swap_locked@8
