diff --git a/buildscript/modules/discoverpython.cmd b/buildscript/modules/discoverpython.cmd
index 92ff303..1374ced 100644
--- a/buildscript/modules/discoverpython.cmd
+++ b/buildscript/modules/discoverpython.cmd
@@ -2,9 +2,6 @@
 @REM Try locating all Python versions via Python Launcher.
 @SET pythonloc=python.exe
 
-@REM Disable Python launcher support until we get some Meson working.
-@GOTO nopylauncher
-
 @rem Query Python launcher for supported Python versions. Hard fail if none found.
 @set ERRORLEVEL=0
 @where /q py.exe
diff --git a/buildscript/modules/mesa3d.cmd b/buildscript/modules/mesa3d.cmd
index 4c3cbf7..1b7dcf4 100644
--- a/buildscript/modules/mesa3d.cmd
+++ b/buildscript/modules/mesa3d.cmd
@@ -4,7 +4,6 @@
 @echo winflexbison is required to build Mesa3D.
 GOTO skipmesa
 )
-@if %pythonver% GEQ 3 GOTO skipmesa
 @cd %mesa%
 @if NOT EXIST mesa if %gitstate%==0 (
 @echo Fatal: Both Mesa code and Git are missing. At least one is required. Execution halted.
