diff --git a/buildscript/modules/discoverpython.cmd b/buildscript/modules/discoverpython.cmd
index faa9831..a6d8e3e 100644
--- a/buildscript/modules/discoverpython.cmd
+++ b/buildscript/modules/discoverpython.cmd
@@ -2,9 +2,6 @@
 @REM Try locating all Python versions via Python Launcher.
 @SET pythonloc=python.exe
 
-@REM Disable Python launcher support until we get some Meson working.
-@GOTO nopylauncher
-
 @rem Query Python launcher for supported Python versions. Hard fail if none found.
 @set ERRORLEVEL=0
 @where /q py.exe
diff --git a/buildscript/modules/mesa3d.cmd b/buildscript/modules/mesa3d.cmd
index 4908338..8078c89 100644
--- a/buildscript/modules/mesa3d.cmd
+++ b/buildscript/modules/mesa3d.cmd
@@ -12,9 +12,6 @@ GOTO skipmesa
 @GOTO skipmesa
 )
 
-@rem Disable WIP Meson support as it doesn't work yet.
-@if %pythonver% GEQ 3 GOTO skipmesa
-
 @REM Aquire Mesa3D source code if missing and enable S3TC texture cache automatically if possible.
 @set buildmesa=n
 @if %gitstate%==0 echo Error: Git not found. Auto-patching disabled.
