diff --git a/src/gallium/targets/graw-gdi/meson.build b/src/gallium/targets/graw-gdi/meson.build
index c3d3a0aac8eb85086f7a35ad08e37aeb58cb4822..563801cddd81e41769972d373277cfd411e7930d 100644
--- a/src/gallium/targets/graw-gdi/meson.build
+++ b/src/gallium/targets/graw-gdi/meson.build
@@ -32,6 +32,7 @@ libgraw_gdi = shared_library(
   dependencies : [
     dep_ws2_32, idep_mesautil, driver_swrast,
   ],
+  name_prefix : host_machine.system() == 'windows' ? '' : 'lib',  # otherwise mingw will create libgraw.dll
 )
 
 libgraw = libgraw_gdi
diff --git a/src/gallium/targets/graw-null/meson.build b/src/gallium/targets/graw-null/meson.build
index c69542705be4ed8922b912e05e3ecb2ffa0e8873..7e539b044186db9fd90ec609b9717701fabb2d61 100644
--- a/src/gallium/targets/graw-null/meson.build
+++ b/src/gallium/targets/graw-null/meson.build
@@ -32,6 +32,7 @@ libgraw_null = shared_library(
   include_directories : inc_common,
   link_with : libgallium,
   dependencies : idep_mesautil,
+  name_prefix : host_machine.system() == 'windows' ? '' : 'lib',  # otherwise mingw will create libgraw_null.dll
 )
 
 libgraw = libgraw_null
diff --git a/src/gallium/targets/osmesa/meson.build b/src/gallium/targets/osmesa/meson.build
index 7792aa1e2fc18a5e74c75c533fe9c1c3bae1b2bf..db10fe03cf36199ada94564c5eee6446b0561afb 100644
--- a/src/gallium/targets/osmesa/meson.build
+++ b/src/gallium/targets/osmesa/meson.build
@@ -58,6 +58,7 @@ libosmesa = shared_library(
     dep_ws2_32, dep_selinux, dep_thread, dep_clock, dep_unwind,
     driver_swrast, driver_swr,
   ],
+  name_prefix : host_machine.system() == 'windows' ? '' : 'lib',  # otherwise mingw will create libosmesa.dll
   soversion : host_machine.system() == 'windows' ? '' : '8',
   version : '8.0.0',
   install : true,
diff --git a/src/mesa/drivers/osmesa/meson.build b/src/mesa/drivers/osmesa/meson.build
index c479b7401316a5dced60592edf0d060fae8711a4..f9ba2c5ade929d6b4d2ac3260e8c7af5b7922210 100644
--- a/src/mesa/drivers/osmesa/meson.build
+++ b/src/mesa/drivers/osmesa/meson.build
@@ -36,6 +36,8 @@ libosmesa = shared_library(
   link_whole : libglapi_static,
   link_with : [libmesa_classic, osmesa_link_with],
   dependencies : [dep_thread, dep_selinux],
+  name_prefix : host_machine.system() == 'windows' ? '' : 'lib',  # otherwise mingw will create libosmesa.dll
+  soversion : host_machine.system() == 'windows' ? '' : '8',
   version : '8.0.0',
   install : true,
 )
