diff --git a/meson.build b/meson.build
index cdff0312e56..eefb02c4d4b 100644
--- a/meson.build
+++ b/meson.build
@@ -1566,7 +1566,14 @@ endif
 dep_m = cc.find_library('m', required : false)
 
 if host_machine.system() == 'windows'
-  dep_regex = meson.get_compiler('c').find_library('regex', required : false)
+# In MSYS2 at least, regex has dependencies of its own
+  dep_part_regex = dependency('regex', required : false)
+  dep_iconv = cc.find_library('iconv', required : false)
+  if dep_part_regex.found() and dep_iconv.found()
+    dep_regex = declare_dependency(dependencies : [dep_part_regex, dep_iconv])
+  else
+    dep_regex = meson.get_compiler('c').find_library('regex', required : false)
+  endif
   if not dep_regex.found()
     dep_regex = declare_dependency(compile_args : ['-DNO_REGEX'])
   endif
