diff --git a/meson.build b/meson.build
index 7c1cc07fd59b60d9a8cf5edddbebe93548d9ff8f..c62b0c541cefb96ca33af8e0ce48077d8e8882d3 100644
--- a/meson.build
+++ b/meson.build
@@ -1430,13 +1430,21 @@ endif
 # it's not linux and wont
 dep_m = cc.find_library('m', required : false)
 
+dep_regex = null_dep
 if host_machine.system() == 'windows'
-  dep_regex = meson.get_compiler('c').find_library('regex', required : false)
-  if not dep_regex.found()
+  dep_regex = cc.find_library('regex', required : false)
+  if dep_regex.found()
+    # systre provides regex but dependencies must be explicited
+    dep_systre = cc.find_library('systre', required : false)
+    if dep_systre.found()
+      dep_tre = cc.find_library('tre', required : false)
+      dep_intl = cc.find_library('intl' , required : false)
+      dep_iconv = cc.find_library('iconv', required : false)
+      dep_regex = [dep_systre, dep_tre, dep_intl, dep_iconv]
+    endif
+  else
     dep_regex = declare_dependency(compile_args : ['-DNO_REGEX'])
   endif
-else
-  dep_regex = null_dep
 endif
 
 if with_platform_haiku
diff --git a/src/util/meson.build b/src/util/meson.build
index 14cc2123511a23becd8ee9bc640e64a95a7a5e32..2b91cb52bbdcb54681a98f60c19e37d3346ad6db 100644
--- a/src/util/meson.build
+++ b/src/util/meson.build
@@ -219,11 +219,10 @@ idep_mesautil = declare_dependency(
   dependencies : [dep_zlib, dep_clock, dep_thread, dep_atomic, dep_m, dep_valgrind],
 )
 
-xmlconfig_deps = []
-if not (with_platform_android or with_platform_windows)
-  xmlconfig_deps += dep_expat
-endif
-xmlconfig_deps += dep_regex
+xmlconfig_deps = [
+  dep_expat,
+  dep_regex,
+]
 
 _libxmlconfig = static_library(
   'xmlconfig',
