diff --git a/meson.build b/meson.build
index 07ef92dbff8430b217b33776623c2e50e3464407..03dda9ee3e790f496472e35456d80e7dad2ea3c2 100644
--- a/meson.build
+++ b/meson.build
@@ -1210,7 +1210,7 @@ if host_machine.system() == 'windows'
     endif
   else
     add_project_link_arguments(
-      cc.get_supported_arguments(
+      cc.get_supported_link_arguments(
         '-Wl,--nxcompat',
         '-Wl,--dynamicbase',
         '-static-libgcc',
@@ -1219,7 +1219,7 @@ if host_machine.system() == 'windows'
       language : ['c'],
     )
     add_project_link_arguments(
-      cpp.get_supported_arguments(
+      cpp.get_supported_link_arguments(
         '-Wl,--nxcompat',
         '-Wl,--dynamicbase',
         '-static-libgcc',
