diff --git a/src/gallium/SConscript b/src/gallium/SConscript
index 358c31e..c71ccfd 100644
--- a/src/gallium/SConscript
+++ b/src/gallium/SConscript
@@ -58,7 +58,7 @@ SConscript([
     'targets/graw-null/SConscript',
 ])
 
-if not env['embedded']:
+if not env['embedded'] and not env['gles'] and not env['swr']:
     SConscript([
         'state_trackers/osmesa/SConscript',
         'targets/osmesa/SConscript',
diff --git a/src/mesa/drivers/SConscript b/src/mesa/drivers/SConscript
index f549910..165619a 100644
--- a/src/mesa/drivers/SConscript
+++ b/src/mesa/drivers/SConscript
@@ -1,6 +1,6 @@
 Import('*')
-
-SConscript('osmesa/SConscript')
+if not env['gles'] and not env['swr']:
+    SConscript('osmesa/SConscript')
 
 if env['x11']:
     SConscript('x11/SConscript')
