diff --git a/src/gallium/state_trackers/osmesa/SConscript b/src/gallium/state_trackers/osmesa/SConscript
index f5519f1..8e0da94 100644
--- a/src/gallium/state_trackers/osmesa/SConscript
+++ b/src/gallium/state_trackers/osmesa/SConscript
@@ -15,7 +15,7 @@ if env['platform'] == 'windows':
         'BUILD_GL32', # declare gl* as __declspec(dllexport) in Mesa headers
         'WIN32_LEAN_AND_MEAN', # http://msdn2.microsoft.com/en-us/library/6dwk3a1z.aspx
     ])
-    if not env['gles']:
+    if 1==1:
         # prevent _glapi_* from being declared __declspec(dllimport)
         env.Append(CPPDEFINES = ['_GLAPI_NO_EXPORTS'])
 
diff --git a/src/gallium/targets/osmesa/SConscript b/src/gallium/targets/osmesa/SConscript
index 3df5c50..5fb9a30 100644
--- a/src/gallium/targets/osmesa/SConscript
+++ b/src/gallium/targets/osmesa/SConscript
@@ -2,6 +2,9 @@ Import('*')
 
 env = env.Clone()
 
+if env['gles']:
+     Return()
+
 env.Prepend(CPPPATH = [
     '#src/mapi',
     '#src/mesa',
diff --git a/src/mesa/drivers/osmesa/SConscript b/src/mesa/drivers/osmesa/SConscript
index 064f3da..a918dc5 100644
--- a/src/mesa/drivers/osmesa/SConscript
+++ b/src/mesa/drivers/osmesa/SConscript
@@ -29,7 +29,7 @@ if env['platform'] == 'windows':
         '_GDI32_', # prevent wgl* being declared __declspec(dllimport)
         'BUILD_GL32', # declare gl* as __declspec(dllexport) in Mesa headers
     ])
-    if not env['gles']:
+    if 1==1:
         # prevent _glapi_* from being declared __declspec(dllimport)
         env.Append(CPPDEFINES = ['_GLAPI_NO_EXPORTS'])
 
