diff --git a/scons/custom.py b/scons/custom.py
index 09946fa73246c1956e09e08a616be6a584dcb6fa..0d4e843b34a8c643198da463e8ab24d4b4cbb4f6 100644
--- a/scons/custom.py
+++ b/scons/custom.py
@@ -48,7 +48,12 @@ import source_list
 # a path directly. We want to support both, so we need to detect the SCons version,
 # for which no API is provided by SCons 8-P
 
-scons_version = tuple(map(int, SCons.__version__.split('.')))
+# Scons version string has consistently been in this format:
+# MajorVersion.MinorVersion.Patch[alpha/beta.yyyymmdd]
+# so this formula should cover all versions regardless of type
+# stable, alpha or beta.
+# For simplicity alpha and beta flags are removed.
+scons_version = tuple(map(int, SCons.__version__.split('.')[:3]))
 
 def quietCommandLines(env):
     # Quiet command lines
diff --git a/scons/gallium.py b/scons/gallium.py
index efe32e06c6c99fcb8d52d8847e92c99c5282c6ad..3f3d257a657de59fb733838a191a0a0a1176a34a 100755
--- a/scons/gallium.py
+++ b/scons/gallium.py
@@ -311,7 +311,14 @@ def generate(env):
     # Speed up dependency checking.  See
     # - https://github.com/SCons/scons/wiki/GoFastButton
     # - https://bugs.freedesktop.org/show_bug.cgi?id=109443
-    scons_version = distutils.version.StrictVersion(SCons.__version__)
+
+    # Scons version string has consistently been in this format:
+    # MajorVersion.MinorVersion.Patch[alpha/beta.yyyymmdd]
+    # so this formula should cover all versions regardless of type
+    # stable, alpha or beta.
+    # For simplicity alpha and beta flags are removed.
+
+    scons_version = distutils.version.StrictVersion('.'.join(SCons.__version__.split('.')[:3]))
     if scons_version < distutils.version.StrictVersion('3.0.2') or \
        scons_version > distutils.version.StrictVersion('3.0.4'):
         env.Decider('MD5-timestamp')
