diff --git a/buildscript/modules/pythonpackages.cmd b/buildscript/modules/pythonpackages.cmd
index d13dca8..4959635 100644
--- a/buildscript/modules/pythonpackages.cmd
+++ b/buildscript/modules/pythonpackages.cmd
@@ -53,7 +53,7 @@
 @set firstpyinstall=0
 )
 @IF NOT %pypack%==0 IF NOT %pypack%==scons %pythonloc% -W ignore -m pip install -U %pypack%
-@IF %pypack%==scons %pythonloc% -W ignore -m pip install -U scons==3.0.1
+@IF %pypack%==scons %pythonloc% -W ignore -m pip install -U scons
 @IF NOT %pypack%==0 echo.
 @IF %pypack%==pywin32 set /p pywin32com=Do you want to install COM and services support - y/n. You'll be asked for admin privileges:
 @IF %pypack%==pywin32 echo.
@@ -67,7 +67,7 @@
 @set /p pyupd=Install/update python packages (y/n):
 @echo.
 @if /I "%pyupd%"=="y" if EXIST "%LOCALAPPDATA%\pip" RD /S /Q "%LOCALAPPDATA%\pip"
-@if /I "%pyupd%"=="y" for /F "skip=2 delims= " %%m in ('%pythonloc% -W ignore -m pip list -o --disable-pip-version-check') do @IF /I NOT "%%m"=="scons" %pythonloc% -W ignore -m pip install -U "%%m"
+@if /I "%pyupd%"=="y" for /F "skip=2 delims= " %%m in ('%pythonloc% -W ignore -m pip list -o --disable-pip-version-check') do @%pythonloc% -W ignore -m pip install -U "%%m"
 @if /I "%pyupd%"=="y" IF %pythonver%==2 IF NOT EXIST "%windir%\system32\pythoncom27.dll" IF NOT EXIST "%windir%\syswow64\pythoncom27.dll" GOTO locatemeson
 @if /I "%pyupd%"=="y" IF %pythonver%==2 powershell -Command Start-Process "%mesa%\mesa-dist-win\buildscript\modules\pywin32.cmd" -Args "%pythonloc%" -Verb runAs
 
