diff --git a/src/util/meson.build b/src/util/meson.build
index 68ae31f421b20bc01f1d51ae285459d3cb9d923c..12d6cc277367530876764a719b70918bd38113e5 100644
--- a/src/util/meson.build
+++ b/src/util/meson.build
@@ -287,8 +287,10 @@ if with_tests
     dependencies : idep_mesautil,
     c_args : [c_msvc_compat_args],
   )
-  if (host_machine.system() == 'windows' and cc.get_id() == 'gcc')
-    # This conversion is only required on mingw
+  if (host_machine.system() == 'windows' and cc.get_id() == 'gcc' and find_program('winepath', required : false).found())
+    # This conversion is only required on mingw crosscompilers, otherwise we hit at least one of these issues
+    # https://gitlab.freedesktop.org/mesa/mesa/-/issues/2690
+    # https://gitlab.freedesktop.org/mesa/mesa/-/issues/2788
     process_test_exe_full_path = run_command(
       'winepath', '-w', process_test_exe.full_path()
     ).stdout().strip()
