diff --git a/common.py b/common.py
index f4f2bb4..be3ccfc 100644
--- a/common.py
+++ b/common.py
@@ -105,9 +105,6 @@ def AddOptions(opts):
     opts.Add(BoolOption('debug', 'DEPRECATED: debug build', 'yes'))
     opts.Add(BoolOption('profile', 'DEPRECATED: profile build', 'no'))
     opts.Add(BoolOption('quiet', 'DEPRECATED: profile build', 'yes'))
-    opts.Add(BoolOption('texture_float',
-                        'enable floating-point textures and renderbuffers',
-                        'no'))
     opts.Add(BoolOption('swr', 'Build OpenSWR', 'no'))
     if host_platform == 'windows':
         opts.Add('MSVC_VERSION', 'Microsoft Visual C/C++ version')
